flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/WorkerResourceSpec.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.resourcemanager;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.api.common.resources.CPUResource;
+import org.apache.flink.configuration.MemorySize;
+import org.apache.flink.runtime.clusterframework.TaskExecutorProcessSpec;
+import org.apache.flink.util.Preconditions;
+
+import java.util.Objects;
+
+/**
+ * Resource specification of a worker, mainly used by SlotManager requesting from ResourceManager.
+ */
+public class WorkerResourceSpec {
+	private final CPUResource cpuCores;
+
+	private final MemorySize taskHeapSize;
+
+	private final MemorySize taskOffHeapSize;
+
+	private final MemorySize networkMemSize;
+
+	private final MemorySize managedMemSize;
+
+	public WorkerResourceSpec(
+		CPUResource cpuCores,
+		MemorySize taskHeapSize,
+		MemorySize taskOffHeapSize,
+		MemorySize networkMemSize,
+		MemorySize managedMemSize) {
+
+		this.cpuCores = Preconditions.checkNotNull(cpuCores);
+		this.taskHeapSize = Preconditions.checkNotNull(taskHeapSize);
+		this.taskOffHeapSize = Preconditions.checkNotNull(taskOffHeapSize);
+		this.networkMemSize = Preconditions.checkNotNull(networkMemSize);
+		this.managedMemSize = Preconditions.checkNotNull(managedMemSize);
+	}
+
+	@VisibleForTesting
+	public WorkerResourceSpec(
```

Comment:
Share we have a inner Builder to reduce multiple testing purpose constructor override?